src/lsm/table_mutable.zig

```diff
@@ -59,19 +59,20 @@ pub fn TableMutableType(comptime Table: type) type {
 
             // The hash map's load factor may allow for more capacity because of rounding:
             assert(table.values.count() <= table.value_count_max);
-            table.dirty = true;
         }
 
         pub fn remove(table: *TableMutable, value: *const Value) void {
             // If the key is already present in the hash map, the old key will not be overwritten
             // by the new one if using e.g. putAssumeCapacity(). Instead we must use the lower
             // level getOrPut() API and manually overwrite the old key.
-            const tombstone = tombstone_from_key(key_from_value(value));
-            const gop = table.values.getOrPutAssumeCapacity(tombstone);
-            gop.key_ptr.* = tombstone;
+            const gop = table.values.getOrPutAssumeCapacity(value.*);
+            gop.key_ptr.* = tombstone_from_key(key_from_value(value));
```

Comment:
Much better!

What can we rename `gop` to? `entry`?

Reply:
Or `upsert`?

Reply:
Zig calls it a `GetOrPutResult`, but `result` is too general. I like `upsert` — very specific, and we don't use the term anywhere else.